pkg/controller/ipam/antrea_ipam_controller.go

```diff
@@ -72,28 +79,36 @@ func NewAntreaIPAMController(crdClient versioned.Interface,
 	podInformer coreinformers.PodInformer) *AntreaIPAMController {
 	c := &AntreaIPAMController{
 		crdClient:               crdClient,
-		namespaceInformer:       namespaceInformer,
+		statefulSetCleanupQueue: workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "statefulSetCleanup"),
+		statefulSetMap:          sync.Map{},
 		namespaceLister:         namespaceInformer.Lister(),
 		namespaceListerSynced:   namespaceInformer.Informer().HasSynced,
-		podInformer:             podInformer,
 		podLister:               podInformer.Lister(),
 		podListerSynced:         podInformer.Informer().HasSynced,
 		statefulSetInformer:     statefulSetInformer,
-		statefulSetLister:       statefulSetInformer.Lister(),
 		statefulSetListerSynced: statefulSetInformer.Informer().HasSynced,
 	}
 
 	// Add handlers for Stateful Set events.
-	//klog.V(2).InfoS("Subscribing for StatefulSet notifications", "controller", controllerName)
-	klog.InfoS("Subscribing for StatefulSet notifications", "controller", controllerName)
+	klog.V(2).InfoS("Subscribing for StatefulSet notifications", "controller", controllerName)
 	statefulSetInformer.Informer().AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
-			DeleteFunc: c.deleteStatefulSet,
+			DeleteFunc: c.enqueueStatefulSetDeleteEvent,
 		},
 	)
 	return c
 }
 
+// Enqueue the STatefulSet delete notification to process by the worker
+func (c *AntreaIPAMController) enqueueStatefulSetDeleteEvent(obj interface{}) {
```

Comment:
@jianjuns Do you think we need to handle StatefulSet update event to release reserved IPs when scale down?
In #3042 , agent will reserve IPs always for StatefulSet Pods and won't release them.

Reply:
Ideally yes, but we can do in a following PR or next release too.

Reply:
Got it.